### PR TITLE
fix special automate/delivery code to handle 'latest' version

### PR DESCRIPTION
**Pull request by chef-delivery**
This Pull Request was opened by Chef Automate user Patrick Wright

Signed-off-by: Patrick Wright <patrick@chef.io>

Ready to merge? View this change in Chef Automate and click the Approve button: https://delivery.chef.co/e/chef/#/organizations/CIA/projects/omnitruck/changes/60a9c172-8cf7-4dc3-9c9e-fffd0756fdeb

### CODE REVIEW

**Comment by wrightp** [comment]
This PR fixes automate and delivery requests that use the 'latest' keyword version param. I didn't know that was even supported :/

I've gone through several refactors trying to limit any changes to VersionResolver (I managed to not change anything)

- The code is overall optimized so all projects other than automate and delivery take fewer code paths
- automate and delivery code paths are slightly more complicated, but for the greater good.

Optimizations to VersionResolver would be great, but not worth the risk at this time.

@schisamo @sersut @chef/engineering-services

File: app.rb [hunk 1]
```diff
@@ -327,22 +327,40 @@ def get_package_info
...
-        current_project = 'delivery' if Mixlib::Versioning.parse(params['v']) < Mixlib::Versioning.parse('0.7.0')
+    if %w{automate delivery}.include?(project)
+      current_version = current_version_resolver.target_version
+
+      # default project
+      current_project = 'automate'
+
+      # If current version is nil we assume latest - automate. Otherwise, check which project it should be
+      if current_version
+        current_project = 'delivery' if current_version < Opscode::Version.parse('0.7.0')
```
Comment (schisamo) [inline app.rb]: I would probably just do all of this in one line:

```ruby
current_project = 'delivery' if current_version && (current_version < Opscode::Version.parse('0.7.0'))
```

Comment (wrightp) [inline app.rb]: Thanks. I had to do it this current way before because of how the version was parsed.

Comment (wrightp) [inline app.rb]: nm. now I know why I did it this way. nil > Opscode::Version fails.  Needs to stay this way

**Pull request merged**